plan/logical_plan_builder.go

```diff
@@ -1790,19 +1790,23 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(columns))...)
 	for i, col := range columns {
 		ds.Columns = append(ds.Columns, col.ToInfo())
-		schema.Append(&expression.Column{
+		newCol := &expression.Column{
 			FromID:   ds.id,
 			Position: i,
 			DBName:   dbName,
 			TblName:  tableInfo.Name,
 			ColName:  col.Name,
 			ID:       col.ID,
 			RetType:  &col.FieldType,
-		})
+		}
 
 		if tableInfo.PKIsHandle && mysql.HasPriKeyFlag(col.Flag) {
-			handleCol = schema.Columns[i]
+			newTp := *newCol.RetType
+			newCol.RetType = &newTp
+			newCol.RetType.Flag |= mysql.UniqueKeyFlag
```

Comment:
Emmm... It's temporary fix. We can't know the column is pk since `histColl` don't have `TableInfo` if don't do this. This will be removed when refactor the planner.

Reply:
The info is cached in `Column`, how can this change affect it?

Reply:
expression is built from the `Column`. So after then column's flag is changed. The column in expression is changed too.

Reply:
Where is it exactly used? Is it here https://github.com/pingcap/tidb/pull/6860/files#diff-20ac3c94a9605bc56ae644dfd45569f8R88?

Reply:
Yes

Reply:
But the info is from cache, it is not changed.